gce/gce.go

```diff
@@ -624,27 +625,58 @@ func (s *gceOps) Expand(
 	volumeID string,
 	newSizeInGiB uint64,
 ) (uint64, error) {
-	// TODO
-	return 0, &cloudops.ErrNotSupported{
-		Operation: "Expand",
+
+	vol, err := s.computeService.Disks.Get(s.inst.project, s.inst.zone, volumeID).Do()
+	if err != nil {
+		return 0, err
 	}
-}
+	currentSizeInGiB := uint64(vol.SizeGb)
 
-func (s *gceOps) Inspect(diskNames []*string) ([]interface{}, error) {
-	allDisks, err := s.getDisksFromAllZones(nil)
+	if currentSizeInGiB >= newSizeInGiB {
+		return currentSizeInGiB, cloudops.NewStorageError(cloudops.ErrDiskGreaterOrEqualToExpandSize,
+			fmt.Sprintf("disk is already has a size: %d greater than or equal "+
+				"requested size: %d", currentSizeInGiB, newSizeInGiB), "")
+	}
+
+	op, err := s.computeService.Disks.Resize(s.inst.project, s.inst.zone, volumeID, &compute.DisksResizeRequest{
+		SizeGb:          int64(newSizeInGiB),
+		ForceSendFields: nil,
+		NullFields:      nil,
+	}).Do()
 	if err != nil {
-		return nil, err
+		return 0, err
 	}
 
-	var disks []interface{}
-	for _, id := range diskNames {
-		if d, ok := allDisks[*id]; ok {
-			disks = append(disks, d)
-		} else {
-			return nil, fmt.Errorf("disk %s not found", *id)
+	// Taken from k8s.io/legacy-cloud-providers/aws
+	backoff := wait.Backoff{
```

Comment:
Lets use the same backoff as defined here: 
https://github.com/kubernetes/legacy-cloud-providers/blob/cebac2e3367faa71a39050bf5563fa7406006e76/gce/gce.go#L869

```
	backoff := wait.Backoff{
		// These values will add up to about a minute. See #56293 for background.
		Duration: time.Second,
		Factor:   1.4,
		Steps:    10,
	}
```